src/game/enemies/madillo.c

```diff
@@ -0,0 +1,200 @@
+#include "global.h"
+#include "malloc_vram.h"
+#include "game/entity.h"
+
+#include "constants/animations.h"
+
+typedef struct {
+    /* 0x00 */ SpriteBase base;
+    /* 0x0C */ Sprite s;
+    /* 0x3C */ s32 spawnX;
+    /* 0x40 */ s32 spawnY;
+    /* 0x44 */ s32 offsetX;
+    /* 0x48 */ s32 offsetY;
+    /* 0x4C */ s32 unk4C;
+    /* 0x50 */ bool8 clampParam;
+    /* 0x51 */ u8 unk51;
+} Sprite_Madillo; /* size: 0x54 */
+
+void Task_MadilloMain(void);
+void Task_8056230(void);
+void Task_80564BC(void);
+void TaskDestructor_80095E8(struct Task *);
+
+void CreateEntity_Madillo(MapEntity *me, u16 spriteRegionX, u16 spriteRegionY,
+                          u8 spriteY)
+{
+    ENTITY_INIT(Sprite_Madillo, madillo, Task_MadilloMain, 0x4040, 0,
+                TaskDestructor_80095E8);
+
+    if (me->d.sData[1] != 0) {
+        madillo->clampParam = TRUE;
+    } else {
+        madillo->clampParam = FALSE;
+    }
+
+    ENEMY_SET_SPAWN_POS_GROUND(madillo, me);
+
+    madillo->unk4C = -Q_24_8(1.5);
+    madillo->unk51 = 120;
+
+    s->x = TO_WORLD_POS(me->x, spriteRegionX);
+    s->y = TO_WORLD_POS(me->y, spriteRegionY);
+    SET_MAP_ENTITY_INITIALIZED(me);
+
+    SPRITE_INIT_EXCEPT_POS(s, 16, SA2_ANIM_MADILLO, 0, 0x480, 2);
+}
+
+// (99.94%) https://decomp.me/scratch/seFRx
+NONMATCH("asm/non_matching/Task_MadilloMain.inc", void Task_MadilloMain(void))
```

Comment:
Took a look and I can't figure it out :(

Reply:
Oh. Whoops :D https://decomp.me/scratch/OTp1Y

Btw, we should 100% be using `Q_24_8_NEW` as there are loads of places where things always match with it like this.

Reply:
There is a difference with how the compiler treats `<< 8` vs `* 256` in some cases

Reply:
> 

How often in this project have I said "WTF I tried that!?" yet? ^^
Good job!